subsys/net/lib/lwm2m/lwm2m_obj_swmgmt.c

```diff
@@ -0,0 +1,647 @@
+#define EVENT_PKG_INTEGRITY_FAILED 8
+#define EVENT_ACTIVATE 9
+#define EVENT_DEACTIVATE 10
+
+/*
+   0: Initial value. Prior to download any new package in the Device, Update
+      Result MUST be reset to this initial value. One side effect of executing
+      the Uninstall resource is to reset Update Result to this initial value 0.
+*/
+#define UPD_RES_INITIAL 0
+
+/* 1: Downloading. The package downloading process is ongoing. */
+#define UPD_RES_DOWNLOADING 1
+
+/* 2: Software successfully installed. */
+#define UPD_RES_SW_SUCCESSFULLY_INSTALLED 2
+
+/* 3: Successfully Downloaded and package integrity verified */
+#define UPD_RES_DOWNLOADED_AND_VERIFIED 3
+
+/* 4-49: reserved for expansion of other scenarios>> */
+/* 50: Not enough storage for the new software package. */
+#define UPD_RES_NOT_ENOUGH_STORAGE 50
+
+/* 51: Out of memory during downloading process. */
+#define UPD_RES_OUT_OF_MEMORY_DURING_DOWNLOAD 51
+
+/* 52: Connection lost during downloading process. */
+#define UPD_RES_LOST_CONNECTION_DURING_DOWNLOAD 52
+
+/* 53: Package integrity check failure. */
+#define UPD_RES_PACKAGE_INTEGRITY_CHECK_FAILURE 53
+
+/* 54: Unsupported package type. */
+#define UPD_RES_UNSUPPORTED_PACKAGE_TYPE 54
+
+/* 55: Undefined */
+/* 56: Invalid URI */
+#define UPD_RES_INVALID_URI 56
+
+/* 57: Device defined update error */
+#define UPD_RES_DEVICE_DEFINED_UPDATE_ERROR 57
+
+/* 58: Software installation failure */
+#define UPD_RES_SW_INSTALLATION_FAILURE 58
+
+/* 59: Uninstallation Failure during forUpdate(arg=0) */
+#define UPD_RES_UNINSTALLATION_FAILURE_FOR_UPDATE 59
+
+/*
+   60-200 : reserved for expansion selection to be in blocks depending on new
+            introduction of features
+*/
+
+struct lwm2m_swmgmt_data {
+	uint8_t obj_inst_id;
+
+	char package_name[PACKAGE_NAME_LEN];
+	char package_version[PACKAGE_VERSION_LEN];
+
+	bool next_package_is_upgrade;
+
+	uint8_t update_state;
+	uint8_t update_result;
+
+	bool activation_state;
+
+	lwm2m_engine_user_cb_t install_package_cb;
+	lwm2m_engine_user_cb_t upgrade_package_cb;
+	lwm2m_engine_user_cb_t delete_package_cb;
+	lwm2m_engine_user_cb_t activate_cb;
+	lwm2m_engine_user_cb_t deactivate_cb;
+	lwm2m_engine_set_data_cb_t write_package_cb;
+
+#ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
+	char package_uri[PACKAGE_URI_LEN];
+	struct firmware_pull_context pull_context;
+#endif
+};
+
+int callback_not_defined(uint16_t obj_inst_id)
+{
+	LOG_ERR("Callback not defined for inst %u", obj_inst_id);
+	return -EINVAL;
+}
+static int write_callback_not_defined(uint16_t obj_inst_id, uint16_t res_id,
+			              uint16_t res_inst_id, uint8_t *data, uint16_t data_len,
+			              bool last_block, size_t total_size)
+{
+	LOG_ERR("Callback not defined for inst %u", obj_inst_id);
+	return -EINVAL;
+}
+
+static struct lwm2m_swmgmt_data swmgmt_data[MAX_INSTANCE_COUNT] = { 0 };
+
+static struct lwm2m_swmgmt_data* find_index(uint16_t obj_inst_id)
+{
+	int index;
+	LOG_INF("Finding instance ...");
```

Comment:
I guess you didn't mean to leave these LOG calls in there?

Reply:
Correct